astroplan/core.py

```diff
@@ -25,82 +40,137 @@ class Observer(object):
-            The ambient relative humidity.
+        if relative_humidity is None:
+            relative_humidity = 0
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        self.pressure = pressure
+        self.temperature = temperature
+        self.relative_humidity = relative_humidity
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and longitude is not None):
+            accepted_latlon_types = [str, Quantity]
+            if (type(latitude) in accepted_latlon_types and
+                type(longitude) in accepted_latlon_types):
+                latitude = Latitude(latitude, unit=u.degree)
+                longitude = Longitude(longitude, unit=u.degree)
+
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
+
+        elif isinstance(location, EarthLocation):
+            self.location = location
 
-        Default time zone is UTC.
-        If time zone (local or otherwise) requested, then uses `datetime.datetime`
-        and pytz.timezone to convert from UTC.
+        else:
+            raise TypeError('Observatory location must be specified with '
+                            'either (1) latitude and longitude in degrees as '
+                            'accepted by astropy.coordinates.Latitude and '
+                            'astropy.coordinates.Latitude, or (2) as an '
+                            'instance of astropy.coordinates.EarthLocation.')
 
-        Must first create date_time object with get_date() before using any other 
-        Observer methods, all of which require one as an argument.
+        # Accept various timezone inputs, default to UTC
+        if isinstance(timezone, datetime.tzinfo):
+            self.timezone = timezone
+        elif isinstance(timezone, str) or isinstance(timezone, unicode):
+            self.timezone = pytz.timezone(timezone)
+        else:
+            raise TypeError('timezone keyword should be a string, or an '
+                            'instance of datetime.tzinfo')
+
+    def altaz(self, time, target=None, obswl=None):
+        """
+        Returns an instance of `~astropy.coordinates.SkyCoord` with altitude and
+        azimuth of the `FixedTarget` called `target` at time `time`.
 
         Parameters
         ----------
-        input_date_time : WHAT TYPE IS astropy.time OBJECT?
+        time : `~astropy.time.Time`
+            Astropy time object.
+
+        target : None (default) or `~astroplan.FixedTarget` or `~astropy.coordinates.SkyCoord`
+            Celestial object of interest. If `target`=None, return just the
+            `~astropy.coordinates.AltAz` frame without coordinates.
 
```

Comment:
My strategy was that the `Observer` should really encapsulate just the observatory. Since you could observe multiple targets with the same observatory, or the same target at multiple wavelengths, I think we should keep `obswl` confined to specific calculations like `obs.altaz()`.